protectwise_connector.py

```diff
@@ -89,59 +229,35 @@ def _make_rest_call(self, endpoint, action_result, headers=None, params=None, da
 
         # handle the error in case the caller specified a non-existant method
         if (not request_func):
-            action_result.set_status(phantom.APP_ERROR, PW_ERR_API_UNSUPPORTED_METHOD, method=method)
+            action_result.set_status(phantom.APP_ERROR, PROTECTWISE_ERR_API_UNSUPPORTED_METHOD, method=method)
 
+        self.save_progress("Making {} call to {} endpoint".format(method, endpoint))
         # Make the call
-        try:
-            r = request_func(self._base_url + endpoint,  # The complete url is made up of the base_url, the api url and the endpiont
-                    data=json.dumps(data) if data else None,  # the data, converted to json string format if present, else just set to None
-                    headers=headers,  # The headers to send in the HTTP call
-                    verify=config.get(phantom.APP_JSON_VERIFY, True),  # should cert verification be carried out?
-                    params=params)  # uri parameters if any
-        except Exception as e:
-            return (action_result.set_status(phantom.APP_ERROR, PW_ERR_SERVER_CONNECTION, e), resp_json)
-
-        # self.debug_print('REST url: {0}'.format(r.url))
-
-        if (hasattr(action_result, 'add_debug_data')):
-            action_result.add_debug_data({'r_text': r.text if r else 'r is None'})
-
-        # Try a json parse, since most REST API's give back the data in json,
-        # if the device does not return JSONs, then need to implement parsing them some other manner
-        try:
-            resp_json = r.json()
-        except Exception as e:
-            # r.text is guaranteed to be NON None, it will be empty, but not None
-            msg_string = PW_ERR_JSON_PARSE.format(raw_text=r.text)
-            return (action_result.set_status(phantom.APP_ERROR, msg_string, e), resp_json)
-
-        # Handle any special HTTP error codes here, many devices return an HTTP error code like 204. The requests module treats these as error,
-        # so handle them here before anything else, uncomment the following lines in such cases
-        # if (r.status_code == 201):
-        #     return (phantom.APP_SUCCESS, resp_json)
-
-        # Handle/process any errors that we get back from the device
-        if (200 <= r.status_code <= 399):
-            # Success
-            return (phantom.APP_SUCCESS, resp_json)
-
-        # Failure
-        action_result.add_data(resp_json)
-
-        details = json.dumps(resp_json).replace('{', '').replace('}', '')
-
-        if (r.status_code in exception_error_codes):
-            # Ok to have this http error for this call, return success, the caller will handle the fact that the response is empty
-            return (action_result.set_status(phantom.APP_SUCCESS), resp_json)
-
-        return (action_result.set_status(phantom.APP_ERROR, PW_ERR_FROM_SERVER.format(status=r.status_code, detail=details)), resp_json)
+        for retry in range(self._number_of_retries + 1):
```

Comment:
I think that `retry` is not the best word here, as the first attempt is not really a retry. Maybe we can use `attempt_idx` of sth similar. I believe it will make this code easier to understand
```suggestion
        for attempt_idx in range(self._number_of_retries + 1):
```

Reply:
Agreed, Changed the word.